cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -29,30 +31,30 @@
 public abstract class RowColumns<T> {
   private final byte[] row;
 
-  private final List<byte[]> columns;
-
-  public byte[] getRow() {
-    return row;
-  }
-
-  public List<byte[]> getColumns() {
-    return columns;
-  }
+  private List<byte[]> columns;
 
   // key & column as byte[]
 
   public RowColumns(byte[] row) {
     this.row = row;
-    this.columns = new ArrayList<>();
   }
 
   public RowColumns(byte[] row, byte[]... columns) {
-    this(row);
-    add(columns);
+    this.row = row;
+    this.columns = Arrays.asList(columns);
+  }
+
+  @SuppressWarnings("unchecked")
+  public T add(byte[] firstColumn, byte[]... moreColumns) {
```

Comment:
If user doesn't specify columns (a second parameter) in constructor, user will be using the constructor with only rowkey.